osu.Game.Tests/Visual/SongSelectV2/TestSceneSongSelectFiltering.cs

```diff
@@ -0,0 +1,336 @@
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Cursor;
+using osu.Framework.Input;
+using osu.Framework.Platform;
+using osu.Framework.Screens;
+using osu.Framework.Testing;
+using osu.Game.Beatmaps;
+using osu.Game.Configuration;
+using osu.Game.Database;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Overlays;
+using osu.Game.Overlays.Toolbar;
+using osu.Game.Rulesets;
+using osu.Game.Rulesets.Mania.Mods;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Osu;
+using osu.Game.Rulesets.Osu.Mods;
+using osu.Game.Screens;
+using osu.Game.Screens.Footer;
+using osu.Game.Screens.Menu;
+using osu.Game.Screens.Select.Filter;
+using osu.Game.Screens.SelectV2;
+using osu.Game.Tests.Resources;
+
+namespace osu.Game.Tests.Visual.SongSelectV2
+{
+    public partial class TestSceneSongSelectFiltering : ScreenTestScene
+    {
+        private BeatmapManager manager = null!;
+        private RulesetStore rulesets = null!;
+        private MusicController music = null!;
+        private OsuConfigManager config = null!;
+
+        private SoloSongSelect songSelect = null!;
+        private BeatmapCarousel carousel => songSelect.ChildrenOfType<BeatmapCarousel>().Single();
+
+        private FilterControl filter => songSelect.ChildrenOfType<FilterControl>().Single();
+        private ShearedFilterTextBox filterTextBox => songSelect.ChildrenOfType<ShearedFilterTextBox>().Single();
+        private int filterOperationsCount;
+
+        [Cached]
+        private readonly ScreenFooter screenFooter;
+
+        [Cached]
+        private readonly OsuLogo logo;
+
+        [Cached(typeof(INotificationOverlay))]
+        private readonly INotificationOverlay notificationOverlay = new NotificationOverlay();
+
+        public TestSceneSongSelectFiltering()
+        {
+            Children = new Drawable[]
+            {
+                new PopoverContainer
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Children = new Drawable[]
+                    {
+                        new Toolbar
+                        {
+                            State = { Value = Visibility.Visible },
+                        },
+                        screenFooter = new ScreenFooter
+                        {
+                            OnBack = () => Stack.CurrentScreen.Exit(),
+                        },
+                        logo = new OsuLogo
+                        {
+                            Alpha = 0f,
+                        },
+                    },
+                },
+            };
+
+            Stack.Padding = new MarginPadding { Top = Toolbar.HEIGHT };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(GameHost host)
+        {
+            RealmDetachedBeatmapStore beatmapStore;
+
+            // These DI caches are required to ensure for interactive runs this test scene doesn't nuke all user beatmaps in the local install.
```

Comment:
I feel like we are better than this in newer tests? May be worth seeing if this can be cleaned up (wouldn't give it more than 15 minutes though).